string source = rogr
string target = camera1
time timestamp
